File: src-tauri/src/domain/task.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Task {
    pub id: Option<i64>,
    pub title: String,
    pub description: Option<String>,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
    pub allocated_minutes: Option<i64>,
    pub status: String,
    pub created_at: Option<String>,
}

File: src/types/task.ts
export type TaskStatus = "planned" | "active" | "done";

export interface Task {
  id?: number;
  title: string;
  description?: string;
  start_date?: string;
  end_date?: string;
  allocated_minutes?: number;
  status: TaskStatus;
  created_at?: string;
}
